cli/azd/pkg/tools/azcli/spring_app.go

```diff
@@ -0,0 +1,250 @@
+		return nil, err
+	}
+
+	springApp, err := client.Get(ctx, resourceGroup, instanceName, appName, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed retrieving spring app properties: %w", err)
+	}
+
+	return &SpringAppProperties{
+		Fqdn: []string{*springApp.Properties.Fqdn},
+	}, nil
+}
+
+func (ss *springService) UploadSpringArtifact(
+	ctx context.Context,
+	subscriptionId, resourceGroup, instanceName, appName, artifactPath string,
+) (*string, error) {
+	credential, err := ss.credentialProvider.CredentialForSubscription(ctx, subscriptionId)
+	if err != nil {
+		return nil, err
+	}
+
+	options := clientOptionsBuilder(ss.httpClient, ss.userAgent).BuildArmClientOptions()
+
+	file, err := os.Open(artifactPath)
+	if err != nil {
+		return nil, fmt.Errorf("artifact do not exists: %s", artifactPath)
+	}
+	defer file.Close()
+
+	springClient, err := armappplatform.NewAppsClient(subscriptionId, credential, options)
+	if err != nil {
+		return nil, fmt.Errorf("creating SpringApps client: %w", err)
+	}
+	storageInfo, err := springClient.GetResourceUploadURL(ctx, resourceGroup, instanceName, appName, nil)
+	if err != nil {
+		return nil, fmt.Errorf("get resource upload URL failed: %w", err)
+	}
+
+	url, _ := url.Parse(*storageInfo.UploadURL)
+
+	fileURL := azfile.NewFileURL(*url, azfile.NewPipeline(azfile.NewAnonymousCredential(), azfile.PipelineOptions{}))
+	err = azfile.UploadFileToAzureFile(ctx, file, fileURL,
+		azfile.UploadToAzureFileOptions{
+			Parallelism: 3,
+			Metadata: azfile.Metadata{
+				"createdby": "AZD",
+			},
+		})
+
+	if err != nil {
+		return nil, fmt.Errorf("artifact '%s' upload failed: %w", artifactPath, err)
+	}
+
+	return storageInfo.RelativePath, nil
+}
+
+func (ss *springService) DeploySpringAppArtifact(
+	ctx context.Context,
+	subscriptionId string,
+	resourceGroup string,
+	instanceName string,
+	appName string,
+	relativePath string,
+	deploymentName string,
+) (*string, error) {
+	deploymentClient, err := ss.createSpringAppDeploymentClient(ctx, subscriptionId)
+	if err != nil {
+		return nil, err
+	}
+
+	springClient, err := ss.createSpringAppClient(ctx, subscriptionId)
+	if err != nil {
+		return nil, err
+	}
+
+	poller, err := deploymentClient.BeginCreateOrUpdate(ctx, resourceGroup, instanceName, appName, deploymentName,
+		armappplatform.DeploymentResource{
+			Properties: &armappplatform.DeploymentResourceProperties{
+				Source: &armappplatform.JarUploadedUserSourceInfo{
+					Type:         to.Ptr("Jar"),
+					RelativePath: to.Ptr(relativePath),
+				},
+			},
+		}, nil)
+
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = springClient.BeginSetActiveDeployments(ctx, resourceGroup, instanceName, appName,
+		armappplatform.ActiveDeploymentCollection{
+			ActiveDeploymentNames: []*string{
+				to.Ptr(deploymentName),
+			},
+		}, nil)
+
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := poller.PollUntilDone(ctx, nil)
+	if err != nil {
+		log.Fatalf("failed to pull the result: %v", err)
```

Comment:
Is `log.FatalF` intentional here? This logs then panics. Did you need to return `err`?